rootfs/api/models.py

```diff
@@ -161,12 +151,30 @@ class Meta:
         permissions = (('use_app', 'Can use app'),)
 
     @property
+    def select_app_name(self):
+        """Select a unique randomly generated app name"""
+        name = utils.generate_app_name()
+
+        while App.objects.filter(id=name).exists():
+            name = utils.generate_app_name()
+
+        return name
+
+    @property
     def _scheduler(self):
         mod = importlib.import_module(settings.SCHEDULER_MODULE)
         return mod.SchedulerClient(settings.SCHEDULER_URL,
                                    settings.SCHEDULER_AUTH,
                                    settings.SCHEDULER_OPTIONS)
 
+    def save(self, **kwargs):
+        if not self.id:
```

Comment:
This is due to the new App Loading logic - No model can used as part of the import until they have been init'ed